--- a/configure.ac
+++ b/configure.ac
@@ -1,14 +1,16 @@
 dnl configure.ac for netatalk
 
-AC_INIT(etc/afpd/main.c)
+AC_CONFIG_SRCDIR([etc/afpd/main.c])
 
-NETATALK_VERSION=`cat $srcdir/VERSION`
+NETATALK_VERSION=m4_esyscmd([cat VERSION])
 AC_SUBST(NETATALK_VERSION)
+AC_INIT([netatalk], NETATALK_VERSION)
 
 AC_CANONICAL_SYSTEM
-AM_INIT_AUTOMAKE(netatalk, ${NETATALK_VERSION})
+AM_INIT_AUTOMAKE([subdir-objects])
 AM_CONFIG_HEADER(config.h)
 AM_MAINTAINER_MODE([enable])
+AC_CONFIG_MACRO_DIRS([macros])
 
 dnl Checks for programs.
 AC_PROG_AWK
--- a/macros/netatalk.m4
+++ b/macros/netatalk.m4
@@ -1,7 +1,7 @@
 dnl Kitchen sink for configuration macros
 
 dnl Check for docbook
-AC_DEFUN(AX_CHECK_DOCBOOK, [
+AC_DEFUN([AX_CHECK_DOCBOOK], [
   # It's just rude to go over the net to build
   XSLTPROC_FLAGS=--nonet
   DOCBOOK_ROOT=
